saraheisa/Photons
# Issue#7 | Connect to DB locally & on Amazon RDS

## Description

This PR creates a successful connection with PostgreSQL DB using [pg driver](https://www.npmjs.com/package/pg)

### Why PostgreSQL?

A relational database is preferred to this application because we have a restrict relation between images and users and we want to guarantee that  every image has a user plus we will need to do some merges to get user data when returning each image

## Related Issue

#7  

## Acceptance Criteria

- [X] Successfully connect to local DB
- [X] Read & write some data to make sure everything is working

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|  ✓ | :link: Update dependencies |
|    | :scroll: Docs              |


## Requests / Responses

**Request**

GET `/images` Returns DB object

**Response**

HTTP/1.1 200 OK

```json
{
    "connectedDB": {
        "_events": {},
        "_eventsCount": 0,
        "options": {
            "user": "postgres",
            "host": "localhost",
            "database": "photons-app-db",
            "port": "5432",
            "max": 10,
            "maxUses": null,
            "idleTimeoutMillis": 10000
        },
        "_clients": [],
        "_idle": [],
        "_pendingQueue": [],
        "ending": false,
        "ended": false
    }
}
```

# Testing Steps
- Install PostgreSql if you don't already have it
- Create a new DB called `photons-app-db` 
- From your terminal, pull down this branch with git pull origin `issue7-Connect-to-db`
- Check that branch out with `git checkout issue7-Connect-to-db`
- Run `npm install` to install dependencies
- Open `server/utilities/DBController.js` and change any of the configurations if needed (you may need to change the password, username or port)
- Run `npm run dev` to run the app
- Head to GET: [`http://localhost:3000/images`](http://localhost:3000/images) form postman or your browser
- Now you should see a JSON object like the one above
